Src/FluentAssertions/Primitives/StringAssertions.cs

```diff
@@ -1114,6 +1114,86 @@ public AndConstraint<TAssertions> BeNullOrWhiteSpace(string because = "", params
             return new AndConstraint<TAssertions>((TAssertions)this);
         }
 
+        /// <summary>
+        /// Asserts that a string is uppercase.
+        /// </summary>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public AndConstraint<TAssertions> BeUpper(string because = "", params object[] becauseArgs)
+        {
+            Execute.Assertion
+                .ForCondition(Subject?.All(char.IsUpper) == true)
+                .BecauseOf(because, becauseArgs)
+                .FailWith("Expected {context:string} to be upper{reason}, but found {0}.", Subject);
```

Comment:
I would be more specific, e.g. `Expected all characters in {context:string} to be upper cased`